pkg/antctl/raw/supportbundle/command.go

```diff
@@ -168,29 +168,45 @@ func request(component string, client *rest.RESTClient) error {
 	return err
 }
 
-func mapClients(prefix string, agentClients map[string]*rest.RESTClient, controllerClient *rest.RESTClient, bar *pb.ProgressBar, af, cf func(nodeName string, c *rest.RESTClient) error) error {
+type result struct {
+	nodeName string
+	err      error
+}
+
+func mapClients(prefix string, agentClients map[string]*rest.RESTClient, controllerClient *rest.RESTClient, bar *pb.ProgressBar, af, cf func(nodeName string, c *rest.RESTClient) error) []result {
 	bar.Set("prefix", prefix)
 	rateLimiter := rate.NewLimiter(requestRate, requestBurst)
+	ch := make(chan result)
 	g, ctx := errgroup.WithContext(context.Background())
 	for nodeName, client := range agentClients {
 		rateLimiter.Wait(ctx)
 		nodeName, client := nodeName, client
 		g.Go(func() error {
 			defer bar.Increment()
-			return af(nodeName, client)
+			err := af(nodeName, client)
+			ch <- result{nodeName: nodeName, err: err}
+			return err
 		})
 	}
-	if err := g.Wait(); err != nil {
-		return err
+
+	results := make([]result, len(agentClients))
+	for i := range results {
+		results[i] = <-ch
 	}
+
+	g.Wait()
+
 	if controllerClient != nil {
 		defer bar.Increment()
-		return cf("", controllerClient)
+		err := cf("", controllerClient)
+		// use "*" so we can use it as a key for map
+		result := result{nodeName: "*", err: err}
```

Comment:
Since it uses `""` as the nodeName when calling `cf`, use it as the nodeName of result to keep consistency?

Reply:
weird it's valid to use "" as key...

Reply:
map won't care whether the key is an empty value. Regardless of the hash algorithm, it can get a hash value for "".

Reply:
Thanks!